src/js/_enqueues/admin/site-icon.js

```diff
@@ -1,138 +1,180 @@
+ *
+ * @since 6.5.0
+ *
+ * @param {string} url The URL of the site icon image.
+ */
+function switchToUpdate( url ) {
+	var chooseButton = jQuery( '#choose-from-library-button' );
+	var iconPreview = jQuery( '#site-icon-preview' );
+	// Set site-icon-img src to the url and remove the hidden class.
+	iconPreview.find( 'img' ).not( '.browser-preview' )
+		.each( function( i, img ) {
+			jQuery(img).attr( 'src', url );
+		});
+	iconPreview.removeClass( 'hidden' );
+	// Remove hidden class from remove.
+	jQuery( '#js-remove-site-icon' ).removeClass( 'hidden' );
+	// If the button is not in the update state, swap the classes.
+	if ( chooseButton.attr( 'data-state' ) !== '1' ) {
+		chooseButton.attr({
+			'class': chooseButton.attr( 'data-alt-classes' ),
+			'data-alt-classes': chooseButton.attr( 'class' ),
+			'data-state': '1'
 		});
-		$( '#site-icon-preview' ).removeClass( 'hidden' );
-		// Remove hidden class from remove.
-		$( '#js-remove-site-icon' ).removeClass( 'hidden' );
-		// If the button is not in the update state, swap the classes.
-		if( $( '#choose-from-library-link' ).attr( 'data-state' ) !== '1' ){
-			var classes = $( '#choose-from-library-link' ).attr( 'class' );
-			$( '#choose-from-library-link' ).attr( 'class', $( '#choose-from-library-link' ).attr('data-alt-classes') );
-			$( '#choose-from-library-link' ).attr( 'data-alt-classes', classes );
-			$( '#choose-from-library-link' ).attr( 'data-state', '1' );
-		}
-
-		// swap the text of the button
-		$( '#choose-from-library-link' ).text( $( '#choose-from-library-link' ).attr( 'data-update-text' ) );
 	}
 
+	// Swap the text of the button.
+	chooseButton.text(
+		chooseButton.attr( 'data-update-text' )
+	);
+	chooseButton.trigger( 'focus' );
+}
+
+/**
+ * Initializes the functionality to remove the site icon.
+ *
+ * @since 6.5.0
+ *
+ * @param {jQuery} $ The jQuery object.
+ */
+( function ( $ ) {
+	var chooseButton = $( '#choose-from-library-button' );
 	$( '#js-remove-site-icon' ).on( 'click', function() {
 		$( '#site_icon_hidden_field' ).val( 'false' );
 		$( '#site-icon-preview' ).toggleClass( 'hidden' );
-		$( this ).toggleClass( 'hidden' );
+		$(this).toggleClass( 'hidden' );
 
-		var classes = $( '#choose-from-library-link' ).attr( 'class' );
-		$( '#choose-from-library-link' ).attr( 'class', $( '#choose-from-library-link' ).attr( 'data-alt-classes' ) );
-		$( '#choose-from-library-link' ).attr( 'data-alt-classes', classes );
+		chooseButton.attr({
+			'class': chooseButton.attr( 'data-alt-classes' ),
+			'data-alt-classes': chooseButton.attr( 'class' ),
+			'data-state': ''
+		});
 
 		// Swap the text of the button.
-		$( '#choose-from-library-link' ).text( $( '#choose-from-library-link' ).attr( 'data-choose-text' ) );
-		// Set the state of the button so it can be changed on new icon.
-		$( '#choose-from-library-link' ).attr( 'data-state', '');
+		chooseButton.text( chooseButton.attr( 'data-choose-text' ) );
+		chooseButton.trigger( 'focus' );
```

Comment:
```suggestion
		chooseButton
			.text( chooseButton.attr( 'data-choose-text' ) )
			.trigger( 'focus' );
```

Reply:
Good fix thanks. Moving focus to the choose button seeems the most appropriate thing to do 👍🏽